Enable imgproxy usage

lucasfischer opened the pull request.

lucasfischer commented:
We can configure the environment of imgproxy like documented here: https://docs.imgproxy.net/configuration/options
We can configure how images should get processed inside the URL like documented here: https://docs.imgproxy.net/usage/processing

Currently, we use `rs:fit:400:400:0/f:webp`, which means images get resized into a 400x400 rectangle and converted to WebP.

There is also the option to define presets on the service-side so we don't need to define processing parameters inside the URL.

We use `IMGPROXY_ALLOWED_SOURCES` on the service, so it only processes URLs from `https://cdn.kinopio.club`

pketh commented on src/components/ImageOrVideo.vue:
```diff
@@ -3,14 +3,30 @@ import { reactive, computed, onMounted, onBeforeUnmount, watch, ref, nextTick }
+
 onMounted(() => {
-  state.imageUrl = props.image || props.pendingUploadDataUrl
+  state.imageUrl = transformURL(props.image) || props.pendingUploadDataUrl
```
instead of fixing the width at 400, i wonder if we can define multiple images based on the size of the container

https://developer.mozilla.org/en-US/docs/Web/HTML/Guides/Responsive_images

```<img
  srcset="elva-fairy-480w.jpg 480w, elva-fairy-800w.jpg 800w"
  sizes="(max-width: 600px) 480px,
         800px"
  src="elva-fairy-800w.jpg"
  alt="Elva dressed as a fairy" />
```

never used it before so i'll try it and see if it works

pketh commented on src/components/ImageOrVideo.vue:
```diff
@@ -3,14 +3,30 @@ import { reactive, computed, onMounted, onBeforeUnmount, watch, ref, nextTick }
+    return url
+  }
+  if (url.includes(consts.cdnHost)) {
+    if (window.location.search.includes('keycdn')) { // KeyCDN
```
@lucasfischer when is `(window.location.search.includes('keycdn')` true? what is the use case of this `if`?

lucasfischer replied:
This is just for testing. When you append `?keycdn=1` to any kinopio.club URL it will use KeyCDNs mechanism for image conversion. I would remove it in production.

lucasfischer commented:
It seems like `srcset` is only able to choose images based on the viewport and not the container size. I updated the code to use different image sizes dependent on the container size. I made use of `window.devicePixelRatio` to adjust image sizes for devices that use a `devicePixelRatio` higher than 1.

Additionally I fixed a bug that wouldn't show the image while the image is uploading.


https://github.com/user-attachments/assets/56996a69-8541-47bb-9d46-23093eb40a40
